cmd/soroban-cli/src/commands/snapshot/create.rs

```diff
@@ -287,6 +312,18 @@ impl Cmd {
                     if seen.contains(&key) {
                         continue;
                     }
+
+                    // Extract TTL settings from StateArchival config entries before filtering
+                    let Some(val) = val else { continue };
+                    if let LedgerEntryData::ConfigSetting(ConfigSettingEntry::StateArchival(
+                        state_archival,
+                    )) = &val.data
```

Comment:
There's a bug with this logic that old configs could be override new configs, because the config could be in multiple buckets but we should only take the first time we see it. If we implement the logic as per https://github.com/stellar/stellar-cli/pull/2219/files#r2374160865, then this issue should be resolved.